log/README.md

```diff
@@ -92,3 +93,31 @@ This checkpoint was issued by the log known as "Moon Log", the log's size is
 4027504, in the `other data` section a timestamp is encoded as a 64bit hex
 value, and further application-specific data relating to the phase of the moon
 at the point the checkpoint was issued is supplied following that.
+
+## Merging Checkpoints
+
+This checkpoint format allows a checkpoint that has been independently signed by
+multiple identities to be merged, creating a single checkpoint with multiple
+signatures. This is particularly useful for witnessing, where witnesses will
+independently check consistency of the log and produce a counter-signed copy
+containing two signatures: one for the log, and one for the witness.
+
+The ability to merge signatures for the same body is a useful optimization.
+Clients that require N witness signatures will not be required to fetch N checkpoints.
+Instead they can fetch a single checkpoint and confirm it has the N required
+signatures (in addition to the log signature).
+
+Note that this optimization requires the checkpoint _body_ to be byte-equivalent.
+The log signature does not need to be equal; when merging only one of the log's
+signatures over this body will be propagated. The core checkpoint format above
+allows merging for any two consistent checkpoints for the same tree size.
+However, if the `otherdata` extension is used then this can lead to checkpoints
+that cannot be merged, even at the same tree size.
+
+We recommend that log operators using the `otherdata` consider carefully what
+information is included in this. If data is included in `otherdata` that is not
+fixed for a given tree size, then this can easily lead to unmergeable checkpoints.
+The most commonly anticipated cause for this would be including the timestamp at
+which the checkpoint was requested within the `otherdata`. In this case, no two
```

Comment:
I guess the most likely is "timestamp at which the checkpoint was _created_", which would probably be fine so long as new checkpoints aren't being created super frequently, but there could be schemes where each requested checkpoint could be unique.

Reply:
I chose to write it this way because it's easy to assume that for each tree size a single checkpoint is created (this is what serverless does, for example). If we write it as "time the checkpoint was created" then this way of thinking leads to one timestamp per tree size, which is a different problem than the one I'm trying to communicate. Of course, ever changing tree sizes are also a problem, but a different problem than "very sticky tree size but very unstable data in `otherdata`".

The timestamp may be the time it was created, a challenge nonce issued from the client, the time the server handled the request, the time that it was sent for signing... in my view "timestamp at which the checkpoint was requested" covers all of these cases reasonably well.

Reply:
Merging this now so there is some discussion of this in sooner than later. Happy to review PRs to tweak the wording if this is still bothering you after the break.